app/code/Magento/Integration/Model/ResourceModel/Oauth/Token/RequestLog.php

```diff
@@ -58,11 +58,14 @@ protected function _construct()
      */
     public function getFailuresCount($userName, $userType)
     {
+        $date = (new \DateTime())->setTimestamp($this->dateTime->gmtTimestamp());
+        $dateTime = $date->format(\Magento\Framework\Stdlib\DateTime::DATETIME_PHP_FORMAT);
+        
         $select = $this->getConnection()->select();
         $select->from($this->getMainTable(), 'failures_count')
-            ->where('user_name = :user_name AND user_type = :user_type');
+            ->where('user_name = :user_name AND user_type = :user_type AND lock_expires_at > :expiration_time');
 
-        return (int)$this->getConnection()->fetchOne($select, ['user_name' => $userName, 'user_type' => $userType]);
+        return (int)$this->getConnection()->fetchOne($select, ['user_name' => $userName, 'user_type' => $userType, 'expiration_time' => $dateTime]);
```

Comment:
Would seem this is WAY too long - try this to get rid of the obvious error. Not sure why the inherits are playing up ad they are in existing code.

```suggestion
        return (int)$this->getConnection()->fetchOne(
             $select, 
             [
                 'user_name' => $userName, 
                 'user_type' => $userType, 
                 'expiration_time' => $dateTime,
             ]
        );
```